python/dgl/nn/pytorch/gt/degree_encoder.py

```diff
@@ -53,36 +58,32 @@ def __init__(self, max_degree, embedding_dim, direction="both"):
             )
         self.max_degree = max_degree
 
-    def forward(self, g):
+    def forward(self, degrees):
         """
         Parameters
         ----------
-        g : DGLGraph
-            A DGLGraph to be encoded. Graphs with more than one type of edges
-            are not allowed.
+        degrees : Tensor
+            If direction is ``both``, it should be the stack of in degrees and out degrees
+            of the batched graph with zero padding, of shape :math:`(2, B, N)`.
+            Otherwise, it can be the in degrees or out degrees of the batched graph
+            with zero padding, of shape :math:`(B, N)`, where :math:`B` is the
+            batch size of the batched graph, and :math:`N` is the maximum number of nodes.
 
         Returns
         -------
         Tensor
-            Return degree embedding vectors of shape :math:`(N, d)`,
-            where :math:`N` is the number of nodes in the input graph and
-            :math:`d` is :attr:`embedding_dim`.
+            Return degree embedding vectors of shape :math:`(B, N, d)`,
+            where :math:`d` is :attr:`embedding_dim`.
         """
-        if len(g.etypes) > 1:
-            raise DGLError(
-                "The input graph should have no more than one type of edges."
-            )
-
-        in_degree = th.clamp(g.in_degrees(), min=0, max=self.max_degree)
-        out_degree = th.clamp(g.out_degrees(), min=0, max=self.max_degree)
+        degrees = th.clamp(degrees, min=0, max=self.max_degree)
 
```

Comment:
Should we include a check for dimensions of `degrees` in each branch below to avoid silent incorrect encoding?